### PR TITLE
Upgrade to Micronaut 4 snapshot

### PR DESCRIPTION
**Pull request by wetted**
This reopens #608 but needs work to build for Micronaut 4.0.x.

### CODE REVIEW

File: .github/workflows/central-sync.yml [hunk 1]
```diff
@@ -22,8 +22,8 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: '11'
+          distribution: 'adopt'
```
Comment (donbeave) [inline .github/workflows/central-sync.yml:25]: should be `temurin`

File: .github/workflows/publish-snapshot.yml [hunk 2]
```diff
@@ -20,8 +20,8 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: '11'
+          distribution: 'adopt'
```
Comment (donbeave) [inline .github/workflows/publish-snapshot.yml:23]: should be `temurin`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -21,8 +21,8 @@ jobs:
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: '11'
+          distribution: 'adopt'
```
Comment (donbeave) [inline .github/workflows/release.yml:24]: should be `temurin`

**Comment by graemerocher** [comment]
Failing because of https://github.com/micronaut-projects/micronaut-core/pull/8350

**Pull request merged**